wallets/rn_cli_wallet/src/components/components/DiscoverListItem.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import {
+  ActivityIndicator,
+  Alert,
+  Image,
+  Pressable,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import useNotifyClientContext from '@/hooks/useNotifyClientContext';
+
+import projectsData from '@/constants/projects-resposne.json';
+
+import {useTheme} from '@/hooks/useTheme';
+
+type ProjectItem = (typeof projectsData)[0];
+
+type DiscoverListItemProps = {
+  item: ProjectItem;
+};
+
+export default function DiscoverListItem({item}: DiscoverListItemProps) {
+  const {account, subscriptions, notifyClient} = useNotifyClientContext();
+  const Theme = useTheme();
+
+  const [subscribing, setSubscribing] = React.useState(false);
```

Comment:
to keep code consistency, can you use `useState` directly? 

`import React, { useState } from 'react';`

Reply:
same with other hooks.

In another PR we could change the linter so we can remove the `React` import as is not required anymore

Reply:
Yeah, I think I should update those with a named import to make the code consistent. IMO I always prefer namespace import which makes the import management much simpler, and I believe improves the maintainability and movability of the code.

A legendary blog post of Kent C Dods from the early times of React: https://epicreact.dev/importing-react-through-the-ages/

I highly agree with namespace option even though both options have the same output with named ones.

Reply:
Thanks for the post 🙏 

some points to add:


* AFAIK metro doesn't have tree shaking so importing just what we use could improve things (not 100% sure)
but checking import costs (vscode extension) i've found this
<img width="576" alt="Screenshot 2024-02-26 at 17 27 54" src="https://github.com/WalletConnect/react-native-examples/assets/25931366/e8d40316-f9c6-4164-83bb-90daa25c0856">
<img width="509" alt="Screenshot 2024-02-26 at 17 27 43" src="https://github.com/WalletConnect/react-native-examples/assets/25931366/a1a1ca2c-fec6-4780-8191-edc4095bb20d">
<img width="588" alt="Screenshot 2024-02-26 at 17 28 06" src="https://github.com/WalletConnect/react-native-examples/assets/25931366/7dd42113-99d7-4cdd-81e0-89c3df046425">

* I like to stick to the accepted conventions in React/React-native. Check the imports in their docs
  * for example: https://reactnative.dev/docs/switch
  * https://react.dev/learn#using-hooks